### PR TITLE
[torch.compile] remove compilation_context and simplify code

### PR DESCRIPTION
**Pull request by youkaichao**
remove the confusing compilation context, figure out cudagraph batchsizes during initialization of the config.

### CODE REVIEW

File: vllm/worker/model_runner.py [hunk 1]
```diff
@@ -763,7 +753,6 @@ def _use_captured_graph(self,
                             max_decode_seq_len: int,
                             max_encoder_seq_len: int = 0) -> bool:
         return (decode_only and not self.runner.model_config.enforce_eager
-                and batch_size <= _BATCH_SIZES_TO_CAPTURE[-1]
                 and max_decode_seq_len <= self.runner.max_seq_len_to_capture
                 and max_encoder_seq_len <= self.runner.max_seq_len_to_capture
                 and batch_size <= self.runner.max_batchsize_to_capture)
```
Comment (youkaichao) [inline vllm/worker/model_runner.py:758]: `self.runner.max_batchsize_to_capture <= _BATCH_SIZES_TO_CAPTURE[-1]`

File: vllm/config.py [hunk 2]
```diff
@@ -2386,6 +2381,21 @@ def init_during_runtime(self):
...
+_BATCH_SIZE_ALIGNMENT = 8
+# all the token sizes that **can** be captured by cudagraph.
+# they can be arbitrarily large.
+# currently it includes: 1, 2, 4, 8, 16, 24, 32, 40, ..., 8192.
+# the actual sizes to capture will be determined by the model,
+# depending on the model's max_num_seqs.
+# NOTE: get_graph_batch_size needs to be updated if this list is changed.
+_BATCH_SIZES_TO_CAPTURE = [1, 2, 4] + [
+    _BATCH_SIZE_ALIGNMENT * i for i in range(1, 1025)
+]
```
Comment (WoosukKwon) [inline vllm/config.py:2397]: Shouldn't this be tunable? At least the max batch size should be tunable I believe.

Comment (WoosukKwon) [reply vllm/config.py:2397]: Also, how does it work with V1? V1 uses slightly different batch sizes for CUDA graphs.

File: vllm/worker/model_runner.py (hunk 1 above)
Comment (WoosukKwon) [reply vllm/worker/model_runner.py:758]: Sorry, can you please explain why we can skip this line?

File: vllm/config.py (hunk 2 above)
Comment (youkaichao) [reply vllm/config.py:2397]: these are all possibly sizes to capture, and you can see the max is very large, i.e. `8192` .

later, the actual max batchsize will be determined by `max_num_seqs` .

v1 does not use these fields.

File: vllm/config.py [hunk 3]
```diff
@@ -2496,6 +2541,28 @@ def __post_init__(self):
...
+            ]
+        else:
+            batch_size_capture_list = []
+            if self.model_config is not None and \
+                not self.model_config.enforce_eager:
+                batch_size_capture_list = [1, 2, 4
+                                           ] + [i for i in range(8, 513, 8)]
+
+        self.compilation_config.init_with_cudagraph_sizes(
+            batch_size_capture_list)
```
Comment (youkaichao) [inline vllm/config.py:2564]: @WoosukKwon the logic is here.

File: vllm/worker/model_runner.py (hunk 1 above)
Comment (youkaichao) [reply vllm/worker/model_runner.py:758]: the condition `self.runner.max_batchsize_to_capture <= _BATCH_SIZES_TO_CAPTURE[-1]` always holds, and if `batch_size <= self.runner.max_batchsize_to_capture` , `batch_size <= _BATCH_SIZES_TO_CAPTURE[-1]` naturally holds.

this is just redundant code.

Comment (WoosukKwon) [reply vllm/worker/model_runner.py:758]: Got it. Thanks for the explanation. Didn't see L769.

File: vllm/config.py (hunk 2 above)
Comment (WoosukKwon) [reply vllm/config.py:2397]: Got it. Thanks for the explanation.

File: vllm/config.py (hunk 3 above)
Comment (WoosukKwon) [reply vllm/config.py:2564]: I see. Then, shouldn't we remove https://github.com/vllm-project/vllm/blob/4c05edb33ae4ae279421ddf981816d070e8ec37a/vllm/v1/worker/gpu_model_runner.py#L103?

Comment (youkaichao) [reply vllm/config.py:2564]: removed in [af2e5f0](https://github.com/vllm-project/vllm/pull/10838/commits/af2e5f04da2f8f6eeab513fa5b25197ab8cb884b)

**Review by WoosukKwon — APPROVED** [review]
LGTM.

**Pull request merged**